op-batcher/batcher/driver.go

```diff
@@ -18,11 +21,17 @@ import (
 	"github.com/ethereum-optimism/optimism/op-service/txmgr"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/txpool"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/log"
 )
 
-var ErrBatcherNotRunning = errors.New("batcher is not running")
+var (
+	ErrBatcherNotRunning = errors.New("batcher is not running")
+
+	// cancelID is the channel ID used to indicate this transaction is a cancellation transaction only
+	cancelID = derive.ChannelID([derive.ChannelIDLength]byte{0xDE, 0xAD, 0xBE, 0xEF, 0xDE, 0xAD, 0xBE, 0xEF, 0xDE, 0xAD, 0xBE, 0xEF, 0xDE, 0xAD, 0xBE, 0xEF})
```

Comment:
Instead of having a magic cancel channel ID, we could define an augmented type around `txID` like
```go
type txRef struct {
    id txID
    isCancel bool
}
```
and then in the driver change the receipts channel and queue to
```go
	receiptsCh := make(chan txmgr.TxReceipt[txRef])
	queue := txmgr.NewQueue[txRef](l.killCtx, l.Txmgr, l.Config.MaxPendingTransactions)
```
so we can more clearly communicate our intent with this bool field instead of a magic channel id.

Reply:
Was hoping avoid adding another indirection here, and saw that channelID was 128 bits and therefore basically collision proof so opted for using a cahnnelID indicator for cancel.  Don't feel so strongly so happy to change it if you prefer it.

Reply:
this is now done.